src/business_logic/state/cached_state.rs

```diff
@@ -110,17 +133,12 @@ impl<T: StateReader + Clone> StateReader for CachedState<T> {
     }
 
     fn count_actual_storage_changes(&mut self) -> (usize, usize) {
```

Comment:
This one is better @mmsc2 it does not have the problem of the other one

Reply:
noted, for some reason the logic was duplicated. the other version was deleted in favor of this one.